session1/solutions/assignment4/Rakefile

```diff
@@ -1 +1,46 @@
-# assignment 4 solution
+require 'pathname'
+require 'rubygems/package'
+
+desc 'create archive file from a path'
+task :archive do
+
+  archive_ext = "tar.gz"  
+  path = ENV["PATH2TAR"] || "."
+  exclude = ENV["EXCLUDE"] + "," + archive_ext || archive_ext
+
+  exclude_list = exclude.split(',')
+
+  # check if path exist and set the archive file name
+  if Dir.exist?(path) == false then
```

Comment:
again, use only `!Dir.exist?(path)`or better switch the if to check for `Dir.exist?(path)`